src/Controller/WebexController.php

```diff
@@ -0,0 +1,103 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JoliCode\SecretSanta\Controller;
+
+use JoliCode\SecretSanta\Application\WebexApplication;
+use JoliCode\SecretSanta\Exception\AuthenticationException;
+use JoliCode\SecretSanta\Model\ApplicationToken;
+use JoliCode\SecretSanta\Webex\UserExtractor;
+use JoliCode\SecretSanta\Webex\WebexProvider;
+use League\OAuth2\Client\Token\AccessToken;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Routing\RouterInterface;
+
+class WebexController extends AbstractController
+{
+    public function __construct(
+        private string $webexClientId,
+        private string $webexClientSecret,
+        private RouterInterface $router,
+    ) {
+    }
+
+    /**
+     * Ask for Webex authentication and store the AccessToken in Session.
+     */
+    #[Route('/auth/webex', name: 'webex_authenticate', methods: ['GET'])]
+    public function authenticate(Request $request, WebexApplication $webexApplication, UserExtractor $userExtractor): Response
+    {
+        $session = $request->getSession();
+
+        $provider = new WebexProvider([
+            'clientId' => $this->webexClientId,
+            'clientSecret' => $this->webexClientSecret,
+            'redirectUri' => $this->router->generate('webex_authenticate', [], RouterInterface::ABSOLUTE_URL),
+        ]);
+
+        if ($request->query->has('error')) {
+            if ('invalid_scope' === $request->query->get('error')) {
+                throw new AuthenticationException(WebexApplication::APPLICATION_CODE, 'Cannot get the required permissions, are you Webex admin?');
+            }
+
+            return $this->redirectToRoute('homepage');
+        }
+
+        if (!$request->query->has('code')) {
+            // If we don't have an authorization code then get one
+            $options = [
+                'scope' => [
+                    'spark:kms', // This scope is required to give your integration permission to interact with encrypted content (such as messages).
+                    // 'spark:people_read',
+                    //
+                    'spark-admin:people_read',
+                    // 'spark:teams_read',
+                ],
+            ];
+            $authUrl = $provider->getAuthorizationUrl($options);
+
+            $session->set(WebexApplication::SESSION_KEY_STATE, $provider->getState());
+
+            return new RedirectResponse($authUrl);
+        }
+
+        // Check given state against previously stored one to mitigate CSRF attack
+        if (empty($request->query->get('state')) || ($request->query->get('state') !== $session->get(WebexApplication::SESSION_KEY_STATE))) {
```

Comment:
🙏 please, [**never**](https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/) use empty

BTW, the following line does the same job and it simpler


```suggestion
        if ($request->query->get('state') !== $session->get(WebexApplication::SESSION_KEY_STATE) {
```